checkdoc/main.go

```diff
@@ -11,20 +11,17 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+//
+// Deprecated: Please use checkfile with argument --file-name README.md instead.
```

Comment:
What do you think about simply removing `checkdoc`? I guess that you/we can deal with updating the Collector repositories once new version of https://github.com/open-telemetry/opentelemetry-go-build-tools is released. The fix should be easy to solve.

Reply:
Oh I see that it is used in quite a lot of repositories 😄  But I do not see any good reason to remove it straight ahead. The problem will be the same once `checkdoc` is removed.

Reply:
I was planning on doing this in 2 steps, e.g. mark as deprecated for now, once released update repos that use it, and then remove for next release but happy to remove it now if you think it is better 😄

I've also created an issue #349 to track the updates, and happy to do this once this gets released!

Reply:
>  happy to remove it now if you think it is better 

I think it will be more pragmatic 😉  I suggest waiting at least one day before doing any changes as others may disagree. It may be good to ask others for opinions.

Reply:
Do you think I am good to remove `checkdoc` now ?

Reply:
@open-telemetry/collector-approvers Can you take a look please?

Reply:
I think we should do it in two PRs: we can deprecate the `checkdoc` Go module on this PR and remove it in a separate one after one release of checkdoc has been made. Otherwise pkg.go.dev will not show that this tool has been deprecated.

Reply:
> Otherwise pkg.go.dev will not show that this tool has been deprecated.

Hadn't though of this. I think this is a good reason to do it in 2 steps. What do you think @pellared ?

Reply:
Sounds good, in that case will leave it as is and mark this as resolved.